__test__/integration/adapters/collateralized_simple_interest_loan_adapter/runners/return_collateral_runner.ts

```diff
@@ -0,0 +1,301 @@
+    private async increaseTime(newTime: number): Promise<void> {
+        const secondsUntilNewTime = newTime - (await this.web3Utils.getCurrentBlockTime());
+
+        await this.web3Utils.increaseTime(secondsUntilNewTime + 1);
+    }
+
+    private async setBalances(scenario: ReturnCollateralScenario): Promise<void> {
+        // 1. Set up Debtor balances.
+
+        // The debtor has the exact amount of the collateral token as
+        // they are going to set up as collateral.
+        await this.collateralToken.setBalance.sendTransactionAsync(
+            DEBTOR.address,
+            scenario.collateralTerms.collateralAmount,
+            {
+                from: CONTRACT_OWNER.address,
+            },
+        );
+
+        // The debtor has more than enough of the principal token to repay debts.
+        await this.principalToken.setBalance.sendTransactionAsync(
+            DEBTOR.address,
+            scenario.simpleTerms.principalAmount.mul(2),
+            {
+                from: CONTRACT_OWNER.address,
+            },
+        );
+
+        // 2. Set up creditor balances.
+
+        // The creditor has exactly the amount necessary to loan to the debtor,
+        // as well as to pay for the creditor fee.
+        await this.principalToken.setBalance.sendTransactionAsync(
+            CREDITOR.address,
+            scenario.simpleTerms.principalAmount.add(this.debtOrder.creditorFee),
+            {
+                from: CONTRACT_OWNER.address,
+            },
+        );
+
+        // 3. Set up underwriter balances.
+
+        // The underwriter has enough balance to pay the underwriter fee.
+        await this.principalToken.setBalance.sendTransactionAsync(
+            UNDERWRITER.address,
+            this.debtOrder.underwriterFee,
+            {
+                from: CONTRACT_OWNER.address,
+            },
+        );
+    }
+
+    private async setApprovals(scenario: ReturnCollateralScenario): Promise<void> {
+        const { debtor, creditor, principalAmount } = this.debtOrder;
+
+        // The debtor grants the transfer proxy an allowance for moving the collateral.
+        await this.collateralToken.approve.sendTransactionAsync(
+            this.tokenTransferProxy.address,
+            scenario.collateralTerms.collateralAmount,
+            { from: debtor },
+        );
+
+        // The debtor grants the transfer proxy some sufficient allowance for making repayments.
+        await this.principalToken.approve.sendTransactionAsync(
+            this.tokenTransferProxy.address,
+            principalAmount.mul(2),
+            { from: debtor },
+        );
+
+        // The creditor grants allowance of the principal token being loaned,
+        // as well as the creditor fee.
+        await this.principalToken.approve.sendTransactionAsync(
+            this.tokenTransferProxy.address,
+            principalAmount.add(this.debtOrder.creditorFee),
+            { from: creditor },
```

Comment:
In the spirit of this being an integration test, these approvals should all happen using the `tokenAPI`